File: backend/src/main/java/com/coursemanager/service/CourseService.java
package com.coursemanager.service;

import com.coursemanager.dto.request.CourseRequest;
import com.coursemanager.dto.response.CourseResponse;
import com.coursemanager.dto.response.CourseSyllabusResponse;
import com.coursemanager.dto.response.DashboardStatsResponse;

import java.util.List;

public interface CourseService {

    List<CourseResponse> getActiveCourses();

    List<CourseResponse> getDeletedCourses();

    CourseResponse getCourseById(Long id);

    CourseSyllabusResponse getCourseSyllabus(Long id);

    DashboardStatsResponse getDashboardStats();

    CourseResponse createCourse(CourseRequest request);

    CourseResponse updateCourse(Long id, CourseRequest request);

    void softDeleteCourse(Long id);

    void restoreCourse(Long id);

    void reorderCourses(List<Long> orderedIds);
}

File: frontend/src/services/courseService.js
import { request } from './apiClient';

export const courseService = {
  getCourses: () => request('/courses'),
  getDeletedCourses: () => request('/courses/deleted'),
  getCourseById: (id) => request(`/courses/${id}`),
  getCourseSyllabus: (id) => request(`/courses/${id}/syllabus`),
  getDashboardStats: () => request('/courses/dashboard-stats'),
  createCourse: (data) => request('/courses', {
    method: 'POST',
    body: JSON.stringify(data)
  }),
  updateCourse: (id, data) => request(`/courses/${id}`, {
    method: 'PUT',
    body: JSON.stringify(data)
  }),
  deleteCourse: (id) => request(`/courses/${id}`, {
    method: 'DELETE'
  }),
  restoreCourse: (id) => request(`/courses/${id}/restore`, {
    method: 'PUT'
  }),
  reorderCourses: (orderedIds) => request('/courses/reorder', {
    method: 'PUT',
    body: JSON.stringify(orderedIds)
  })
};

export default courseService;
